demo/src/main/java/com/bluelinelabs/conductor/demo/controllers/TransitionDemoController.kt

```diff
@@ -0,0 +1,142 @@
+package com.bluelinelabs.conductor.demo.controllers
+
+import android.content.res.ColorStateList
+import android.os.Bundle
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.annotation.ColorRes
+import androidx.annotation.DimenRes
+import androidx.core.content.ContextCompat
+import androidx.core.os.bundleOf
+import androidx.core.view.isGone
+import androidx.core.view.updateLayoutParams
+import com.bluelinelabs.conductor.Controller
+import com.bluelinelabs.conductor.ControllerChangeHandler
+import com.bluelinelabs.conductor.RouterTransaction
+import com.bluelinelabs.conductor.changehandler.FadeChangeHandler
+import com.bluelinelabs.conductor.changehandler.HorizontalChangeHandler
+import com.bluelinelabs.conductor.changehandler.VerticalChangeHandler
+import com.bluelinelabs.conductor.demo.R
+import com.bluelinelabs.conductor.demo.changehandler.ArcFadeMoveChangeHandler
+import com.bluelinelabs.conductor.demo.changehandler.CircularRevealChangeHandlerCompat
+import com.bluelinelabs.conductor.demo.changehandler.FlipChangeHandler
+import com.bluelinelabs.conductor.demo.controllers.base.BaseController
+import com.bluelinelabs.conductor.demo.databinding.ControllerTransitionDemoBinding
+import com.bluelinelabs.conductor.demo.util.viewBinding
+
+class TransitionDemoController(args: Bundle) : BaseController(R.layout.controller_transition_demo, args) {
+  private val binding: ControllerTransitionDemoBinding by viewBinding(ControllerTransitionDemoBinding::bind)
+
+  override val title = "Transition Demos"
+
+  private val demo = TransitionDemo.values()[args.getInt(KEY_INDEX)]
+
+  constructor(index: Int) : this(bundleOf(KEY_INDEX to index))
+
+  override fun onViewCreated(view: View) {
+    super.onViewCreated(view)
+
+    if (demo.layout == TransactionDemoLayout.SHARED) {
+      binding.bgView.isGone = true
+    }
+
+    binding.title.updateLayoutParams<ViewGroup.MarginLayoutParams> { topMargin = view.resources.getDimension(demo.layout.topMargin).toInt() }
+    binding.next.updateLayoutParams<FrameLayout.LayoutParams> { gravity = demo.layout.fabGravity }
+
+    if (demo.colorId != 0) {
+      binding.bgView.setBackgroundColor(ContextCompat.getColor(activity!!, demo.colorId))
+    }
+
+    val nextIndex = demo.ordinal + 1
+    val buttonColor = if (nextIndex < TransitionDemo.values().size) {
+      TransitionDemo.values()[nextIndex].colorId
+    } else {
+      R.color.blue_grey_300
+    }
+
+    binding.next.backgroundTintList = ColorStateList.valueOf(ContextCompat.getColor(view.context, buttonColor))
```

Comment:
You can just use `getColorStateList`, the framework handles putting a plain color into a CSL

Reply:
TIL